mbed_lstools/lstools_base.py

```diff
@@ -73,6 +73,8 @@ def __init__(self):
         self.lock_file = os.path.join(MbedLsToolsBase.HOME_DIR,
                                       MbedLsToolsBase.MBEDLS_HOME_DIR, MbedLsToolsBase.MBEDLS_GLOBAL_LOCK)
         self.mbedls_get_mocks()
+        # Make sure retargeting is applied if retarget file exists
+        self.retarget(True)
```

Comment:
> IMO, retargeting is meant to be completely transparent to the program relying on mbedls, which leaves control over whether or not to retarget in the hands of the user.

@stevew817 You bring up a good point. Considering we make the same assumption about mocked targets, we should probably do the same with retargeting to keep the experience consistent. Any thoughts here @mazimkhan ?